software/controller/lib/core/v03system.cpp

```diff
@@ -133,6 +144,10 @@ void V03Actuators::execute(const ActuatorsState &desired_state) {
   }
 
   psol_->set(desired_state.fio2_valve);
+
+  desired_state.led_red ? led_red_->set() : led_red_->clear();
+  desired_state.led_yellow ? led_yellow_->set() : led_yellow_->clear();
+  desired_state.led_green ? led_green_->set() : led_green_->clear();
```

Comment:
I'm not convinced that setting the LEDs belongs in `actuators.execute()`: we may want to do stuff with the LEDs that could be asynchronous from the main control loop, much like the buzzer.

Reply:
Whenever we get around to using them, it should be easy enough to change this aspect of the interface. But as instances they do belong in this class. Also, I believe they are somehow not fully functional in v0.3. I recall Edwin saying some of them were places on the board upside down and are not visible. So I'm gonna leave this as is and if/when we need it, we can improve this.